### PR TITLE
A4A Dev Sites: Add development badge to A4A licenses page.

### PR DESCRIPTION
**Pull request by yansern**
Closes https://github.com/Automattic/dotcom-forge/issues/9038

## Proposed Changes

* Add development badge to A4A Licenses page.
* Round the corners of badges on the A4A Licenses page.

## Why are these changes being made?

* See pdtkmj-2Te-p2#comment-5571

## Testing Instructions

1. Apply D161261-code on your sandbox.
2. Go to https://agencies.automattic.com/purchases/licenses.
3. You should see development badge on dev sites.

Screenshot 2024-09-17 at 22 33 20

Markup on 2024-09-13 at 12:34:08

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by andregardi — APPROVED** [review]

**Comment by ivan-ottinger** [comment]
Related discussion on the badge location: p1726214261475279/1726138096.322779-slack-C02TCEHP3HA.
→ We are going to move the badge to a different column.

**Comment by mashikag** [comment]
I asked help on design whether it is desired to round edges for the badges in the view: p1726590650521589-slack-C06JY8QL0TU.

File: client/a8c-for-agencies/data/purchases/lib/format-licenses.ts [hunk 1]
```diff
@@ -38,5 +46,20 @@ export default function formatLicenses( items: APILicense[] ): License[] {
 		quantity: item.quantity,
 		parentLicenseId: item.parent_license_id,
 		referral: item.referral,
+		meta: formatLicenseMeta( item.meta ),
 	} ) );
 }
+
+export function formatLicenseMeta( meta: APILicenseMeta = {} ): LicenseMeta {
+	const isDevSite = meta?.a4a_is_dev_site === '1' ? true : false;
```
Comment (mashikag) [inline client/a8c-for-agencies/data/purchases/lib/format-licenses.ts]: I believe there is no need for the ternary operator here. 
`meta?.a4a_is_dev_site === '1'` will always return a boolean. Same goes for the line below.

**Comment by mashikag** [comment]
I reverted back the changes to round the badges' edges based on: p1726591010404659/1726590650.521589-slack-C06JY8QL0TU.

I also resolved conflicts. There was another type of license badge ("Transferred") introduced in the meantime which caused multiple conflicts: https://github.com/Automattic/wp-calypso/pull/94580.

I tested for regressions to the best of my abilities. I would appreciate if either @tiagonoronha or @andrii-lysenko could test it out for no regressions. 

**UPDATE:** I just noticed that @tiagonoronha has a backend fix in progress to return `null` instead of `[]` for when the meta is empty: D161614-code. This means that I will have to adjust the types to reflect that. I will do it tomorrow morning.

**Comment by mashikag** [comment]
> UPDATE: I just noticed that @tiagonoronha has a backend fix in progress to return null instead of [] for when the meta is empty: D161614-code. This means that I will have to adjust the types to reflect that. I will do it tomorrow morning.

Done. I adjusted the type of `meta` to reflect the change in: D161614-code: 
```
meta: APILicenseMeta | null;
```

**Review by tiagonoronha — APPROVED** [review]
Tested and both the Development and Transferred badge are working as expected. No regressions as far as I can see!

**Pull request merged**